packages/firebase_vertexai/firebase_vertexai/lib/src/live_api.dart

```diff
@@ -0,0 +1,302 @@
+// limitations under the License.
+import 'api.dart';
+import 'content.dart';
+import 'error.dart';
+
+/// Represents the available voice options for speech synthesis.
+enum Voices {
+  // ignore: public_member_api_docs, constant_identifier_names
+  Aoede('Aoede'),
+
+  // ignore: public_member_api_docs, constant_identifier_names
+  Charon('Charon'),
+
+  // ignore: public_member_api_docs, constant_identifier_names
+  Fenrir('Fenrir'),
+
+  // ignore: public_member_api_docs, constant_identifier_names
+  Kore('Kore'),
+
+  // ignore: public_member_api_docs, constant_identifier_names
+  Puck('Puck');
+
+  const Voices(this._jsonString);
+  final String _jsonString;
+
+  // ignore: public_member_api_docs
+  String toJson() => _jsonString;
+}
+
+/// Configures speech synthesis settings.
+class SpeechConfig {
+  /// Creates a [SpeechConfig] instance.
+  ///
+  /// [voice] (optional): The desired voice for speech synthesis.
+  SpeechConfig({this.voice});
+
+  /// The voice to use for speech synthesis.
+  final Voices? voice;
+  // ignore: public_member_api_docs
+  Map<String, Object?> toJson() => {
+        if (voice != null)
+          'voice_config': {
+            'prebuilt_voice_config': {'voice_name': voice!.toJson()}
+          }
+      }; // Or null
+}
+
+/// Represents the available response modalities.
+enum ResponseModalities {
+  /// Unspecified response modality.
+  unspecified('MODALITY_UNSPECIFIED'),
+
+  /// Text response modality.
+  text('TEXT'),
+
+  /// Image response modality.
+  image('IMAGE'),
+
+  /// Audio response modality.
+  audio('AUDIO');
+
+  const ResponseModalities(this._jsonString);
+  final String _jsonString;
+
+  /// Convert to json format
+  String toJson() => _jsonString;
+}
+
+/// Configures live generation settings.
+final class LiveGenerationConfig extends BaseGenerationConfig {
+  // ignore: public_member_api_docs
+  LiveGenerationConfig({
+    this.speechConfig,
+    this.responseModalities,
+    super.candidateCount,
+    super.maxOutputTokens,
+    super.temperature,
+    super.topP,
+    super.topK,
+  });
+
+  /// The speech configuration.
+  final SpeechConfig? speechConfig;
+
+  /// The list of desired response modalities.
+  final List<ResponseModalities>? responseModalities;
+
+  @override
+  Map<String, Object?> toJson() => {
+        ...super.toJson(),
+        if (speechConfig case final speechConfig?)
+          'speechConfig': speechConfig.toJson(),
+        if (responseModalities case final responseModalities?)
+          'responseModalities':
+              responseModalities.map((modality) => modality.toJson()).toList(),
+      };
+}
+
+/// An abstract class representing a message received from a live server.
+///
+/// This class serves as a base for different types of server messages,
+/// such as content updates, tool calls, and tool call cancellations.
+/// Subclasses should implement specific message types.
+abstract class LiveServerMessage {}
```

Comment:
Should this be `sealed`? Do you intend for external authors to write subclasses?

https://dart.dev/language/class-modifiers#sealed